trace_decoder/src/core.rs

```diff
@@ -533,10 +547,147 @@ fn middle<StateTrieT: StateTrie + Clone>(
     Ok(out)
 }
 
+#[allow(unused_variables)]
+/// Performs all the pre-txn execution rules of the targeted network.
+fn do_pre_execution<StateTrieT: StateTrie + Clone>(
```

Comment:
Are you also storing the `BLOCK_INFO_ROOT`?

Reply:
I believe this is only during _post_state_execution_, not _pre_, and following discussions of this morning we may actually not need to write this `BLOCK_INFO_ROOT` (pending design decisions)